### PR TITLE
pyUmbral configuration and usage documentation: Alice, Ursula, and Bob's actions.

### PR DESCRIPTION
**Pull request by KPrasch**
Makes semantic and grammar corrections to documentation and README.  Adds a pyUmbral configuration detail page, and explains the actions of the characters in order to illustrate the problem metaphor in a "using pyUmbral" documentation module.

Depends on PR #71

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    from umbral import umbral, keys
+
+    # Generate a key pair for Alice
+    alices_private_key = keys.UmbralPrivateKey.gen_key()
+    alices_public_key = private_key.get_pubkey()
+
+    # Encrypt data with bob's public key
+    plaintext = b'Proxy Re-encryption is cool!'
+    ciphertext, capsule = umbral.encrypt(alices_public_key, plaintext)
+
```
Comment (cygnusv) [inline README.rst]: Comment should say "Encrypt data with Alice's public key"

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    # Decrypt data with Bob's keys
+    cleartext = umbral.decrypt(capsule, alices_private_key, ciphertext, alices_public_key)
+
```
Comment (cygnusv) [inline README.rst]: Same here

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    capsule = capsule
+
+    # Generate split re-encryption keys with "M of N"
+    # Minimum threshold of 10 ("M") - and 20 total shares ("N").
+    kfrags, _ = umbral.split_rekey(alices_private_key, bobs_public_key, 10, 20)
+
+    # Ursula re-encrypts the shares and bob attaches them to the capsule
+    for kfrag in kfrags:
+        cfrag = umbral.reencrypt(kfrag, umbral_capsule)
+        bob_capsule.attach_cfrag(cfrag)
```
Comment (cygnusv) [inline README.rst]: what`s `umbral_capsule` and `bob_capsule`? There is an issue here with the name of the capsules. I think all should have distinctive names (e.g., alices_capsule, bob_capsule)

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+    # Bob opens the capsule and decrypts the re-encrypted ciphertext
+    plaintext = umbral.decrypt(capsule, bobs_private_key, ciphertext, alices_public_key)
+
```
Comment (cygnusv) [inline README.rst]: another `capsule`, but which one?

File: README.rst [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+Features
+==========
+- Bulk Data and Key Encapsulation
+- Asymmetric Key Pair Generation
+- Public Key Encryption & Decryption
+- Data Re-encryption
+- Re-encryption Key Generation
+- Threshold re-encryption key splitting (via Shamir's Secret Sharing)
+
```
Comment (cygnusv) [inline README.rst]: Perhaps everything should be in the inverse order. I think the PRE features are the most prominent.

File: docs/source/index.rst [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+Features
+=========
+- Bulk Data and Key Encapsulation
+- Asymmetric Key Pair Generation
+- Public Key Encryption & Decryption
+- Data Re-encryption
+- Re-encryption Key Generation
+- Threshold re-encryption key splitting (via Shamir's Secret Sharing)
+
```
Comment (cygnusv) [inline docs/source/index.rst]: Same as above

File: README.rst (hunk 1 above)
Comment (KPrasch) [reply README.rst]: :heavy_check_mark:

File: README.rst (hunk 2 above)
Comment (KPrasch) [reply README.rst]: :heavy_check_mark:

File: README.rst (hunk 3 above)
Comment (KPrasch) [reply README.rst]: Hmm, within the context of this story, isn't it the same capsule?

File: README.rst (hunk 5 above)
Comment (KPrasch) [reply README.rst]: Agreed, :heavy_check_mark:

File: docs/source/index.rst (hunk 6 above)
Comment (KPrasch) [reply docs/source/index.rst]: :heavy_check_mark:

File: README.rst (hunk 3 above)
Comment (cygnusv) [reply README.rst]: oh yes, we have had so many different versions of the capsule concept that I forgot that right now we only have one capsule that gets enriched with cfrags.
OK, then you're right, both should be the same.  Perhaps just `capsule` in both cases?

Comment (KPrasch) [reply README.rst]: :heavy_check_mark:

File: README.rst [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+    capsule = capsule
+
+    # Generate split re-encryption keys with "M of N"
+    # Minimum threshold of 10 ("M") - and 20 total shares ("N").
+    kfrags, _ = umbral.split_rekey(alices_private_key, bobs_public_key, 10, 20)
+
+    # Ursula re-encrypts the shares and bob attaches them to the capsule
+    for kfrag in kfrags:
+        cfrag = umbral.reencrypt(kfrag, capsule)
+        bob_capsule.attach_cfrag(cfrag)
```
Comment (cygnusv) [inline README.rst]: bob_capsule --> capsule

Comment (KPrasch) [reply README.rst]: Ah! Thank You. :heavy_check_mark:

**Comment by KPrasch** [comment]
OK, commits were getting a bit sparse, so I squashed the smaller changes into 9e58d30, which includes all prior requests for changes along with 9fa4ab7.

File: docs/source/index.rst [hunk 8]
```diff
@@ -8,7 +8,7 @@ pyUmbral
 .. image:: https://travis-ci.org/nucypher/pyUmbral.svg?branch=master
    :target: https://travis-ci.org/nucypher/pyUmbral
 
-pyUmbral is a python implementation of David Nuñes's threshold proxy rencryption scheme: Umbral_.
+pyUmbral is a python implementation of David Nuñez's threshold proxy rencryption scheme: Umbral_.
```
Comment (KPrasch) [inline docs/source/index.rst]: Whoops - Sorry! Thank You.

**Comment by tuxxy** [comment]
Can we go ahead and merge this?

**Pull request merged**